octopus_deploy/datadog_checks/octopus_deploy/check.py

```diff
@@ -73,10 +74,31 @@ def _process_endpoint(self, endpoint, params=None, report_service_check=False):
                 self.warning("Failed to access endpoint: %s: %s", endpoint, e)
                 return {}
 
+    def _process_paginated_endpoint(self, endpoint, params=None, report_service_check=False):
+        skip = 0
+        take = self.config.paginated_limit
+        num_pages = 1
+        num_pages_seen = 0
+        all_items = []
+        params = {} if params is None else params
+        while num_pages_seen < num_pages:
+            params['skip'] = skip
+            params['take'] = take
+
+            response_json = self._process_endpoint(endpoint, params=params, report_service_check=report_service_check)
+            if response_json == {}:
+                return response_json
+            items = response_json.get("Items")
+            num_pages_seen += 1
+            num_pages = response_json.get("NumberOfPages", num_pages)
+            skip += self.config.paginated_limit
+            all_items = items + all_items
```

Comment:
Why are items in later pages being added in front of items from earlier pages? Correct me if I'm wrong, but if page 1 had data like [1,2,3] and page 2 had data like [4,5,6], then all_items would be [4,5,6,1,2,3]. Is there a specific purpose of doing this instead of `all_items = all_items + items`?